lua/resty/moesif/moesif_ser.lua

```diff
@@ -338,30 +339,38 @@ function _M.prepare_message(config)
     ngx_log(ngx.DEBUG, "[moesif] response headers before sending to moesif ", dump(response_headers))
   end
 
-  return {
-    request = {
-      uri = ngx.var.scheme .. "://" .. ngx.var.host .. ":" .. ngx.var.server_port .. ngx.var.request_uri,
-      headers = request_headers,
-      body = request_body_entity,
-      verb = req_get_method(),
-      ip_address = client_ip.get_client_ip(request_headers),
-      api_version = api_version,
-      time = os.date("!%Y-%m-%dT%H:%M:%S.", req_start_time()) .. string.format("%d",(req_start_time()- string.format("%d", req_start_time()))*1000),
-      transfer_encoding = req_body_transfer_encoding,
-    },
-    response = {
-      time = os.date("!%Y-%m-%dT%H:%M:%S.", ngx_now()) .. string.format("%d",(ngx_now()- string.format("%d",ngx_now()))*1000),
-      status = ngx.status,
-      ip_address = Nil,
-      headers = response_headers,
-      body = response_body_entity,
-      transfer_encoding = rsp_body_transfer_encoding,
-    },
-    session_token = session_token_entity,
-    user_id = user_id_entity,
-    company_id = company_id_entity,
-    direction = "Incoming"
-  }
+  if request_uri ~= nil then 
+    return {
+      request = {
+        uri = ngx.var.scheme .. "://" .. ngx.var.host .. ":" .. ngx.var.server_port .. request_uri,
```

Comment:
@keyur9 I am assuming the request_uri will not drop query params, if any. Correct?

Reply:
yup, it'll not drop any query params